src/compiler/scala/tools/nsc/javac/JavaScanners.scala

```diff
@@ -702,6 +718,120 @@ trait JavaScanners extends ast.parser.ScannersCommon {
         in.next()
       }
 
+    /** read a triple-quote delimited text block, starting after the first three
+      * double quotes
+      */
+    private def getTextBlock(): Unit = {
+      // Open delimiter is followed by optional space, then a newline
+      while (in.ch == ' ' || in.ch == '\t' || in.ch == FF) {
+        in.next()
+      }
+      if (in.ch != LF && in.ch != CR) { // CR-LF is already normalized into LF by `JavaCharArrayReader`
+        syntaxError("illegal text block open delimiter sequence, missing line terminator")
+        return
+      }
+      in.next()
+
+      /* Do a lookahead scan over the full text block to:
+       *   - compute common white space prefix
+       *   - find the offset where the text block ends
+       */
+      var commonWhiteSpacePrefix = Int.MaxValue
+      var blockEndOffset = 0
+      val backtrackTo = in.copy
+      var blockClosed = false
+      var lineWhiteSpacePrefix = 0
+      var lineIsOnlyWhitespace = true
+      while (!blockClosed && (in.isUnicode || in.ch != SU)) {
+        if (in.ch == '\"') { // Potential end of the block
+          in.next()
+          if (in.ch == '\"') {
+            in.next()
+            if (in.ch == '\"') {
+              blockClosed = true
+              commonWhiteSpacePrefix = commonWhiteSpacePrefix min lineWhiteSpacePrefix
+              blockEndOffset = in.cpos - 2
+            }
+          }
+
+          // Not the end of the block - just a single or double " character
+          if (!blockClosed) {
+            lineIsOnlyWhitespace = false
+          }
+        } else if (in.ch == CR || in.ch == LF) { // new line in the block
+          in.next()
+          if (!lineIsOnlyWhitespace) {
+            commonWhiteSpacePrefix = commonWhiteSpacePrefix min lineWhiteSpacePrefix
+          }
+          lineWhiteSpacePrefix = 0
+          lineIsOnlyWhitespace = true
+        } else if (lineIsOnlyWhitespace &&
+                   (in.ch == ' ' || in.ch == '\t' || in.ch == FF)) { // extend white space prefix
```

Comment:
hmm, maybe here (and for trailing whitespace) a different definition of "white space" applies?

https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/lang/String.html#stripIndent()

refers to

https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/lang/Character.html#isWhitespace(int)

what do you think? at least that method exists on JDK 8, so it would be easy to change.

Reply:
Given that this appears to be what `javac` does, I guess we should too. I'm struggling to see how the implementation of `stripIndent` matches the spec's definition of whitespace https://docs.oracle.com/javase/specs/jls/se16/html/jls-3.html#jls-WhiteSpace.

Reply:
If I understand the spec correctly, the spec's definition of `WhiteSpace` is only used for the whitespace following the initial `"""`. For the content of the text block (where leading and trailing whitespace is handled), the spec delegates to `String.stripIndent` and `String.translateEscapes`.